src/ring_theory/finiteness.lean

```diff
@@ -373,20 +374,19 @@ begin
   obtain ⟨m, I, e, hfg⟩ := iff.1 hfp,
   refine equiv _ (mv_polynomial.map_alg_equiv (fin n) e),
   -- typeclass inference seems to struggle to find this path
-  letI : is_scalar_tower R
-    (_root_.mv_polynomial (fin m) R) (_root_.mv_polynomial (fin m) R) :=
+  letI : is_scalar_tower R (mv_polynomial (fin m) R) (mv_polynomial (fin m) R) :=
       is_scalar_tower.right,
   letI : is_scalar_tower R
-    (_root_.mv_polynomial (fin m) R)
-    (_root_.mv_polynomial (fin n) (_root_.mv_polynomial (fin m) R)) :=
+    (mv_polynomial (fin m) R) (mv_polynomial (fin n) (mv_polynomial (fin m) R)) :=
       mv_polynomial.is_scalar_tower,
 
   refine equiv _ ((@mv_polynomial.quotient_equiv_quotient_mv_polynomial
     _ (fin n) _ I).restrict_scalars R).symm,
-  refine quotient (submodule.map_fg_of_fg I hfg _) _,
+  refine finite_presentation.quotient (submodule.map_fg_of_fg I hfg _) _,
   let := mv_polynomial.sum_alg_equiv R (fin n) (fin m),
   refine equiv _ this,
-  exact equiv (mv_polynomial R (fin (n + m))) (mv_polynomial.rename_equiv R fin_sum_fin_equiv).symm
+  exact (finite_presentation.mv_polynomial R (fin (n + m))).equiv
+    (mv_polynomial.rename_equiv R fin_sum_fin_equiv).symm
```

Comment:
This section was the primary motivation; in #10024 I'm hitting timeouts, and it's annoying that what appears in the goal state doesn't work, because `mv_polynomial` was shadowed by `finite_presentation.mv_polynomial`

Reply:
Thanks for taking this on.